### PR TITLE
create marionette view for account setup

### PR DESCRIPTION
**Pull request by ChristophWurst**
This refactoring has following advantages:
* fewer jquery hacks in ``init.js``
* self-contained marionette view
* ajax communication moved to account service

Things that changed:
* no loading spinner is shown in the middle of all inputs while loading. I always thought it kind of looked odd.

@DeepDiver1975 @Gomez @itheiss @irgendwie @jancborchardt review please :-)

### CODE REVIEW

**Comment by itheiss** [comment]
Hi!

Downloaded latest nightly from 19 October and applied the diff from your PR. After logging in the mail app was still working with my two accounts. So I removed the first account and noticed nothing happens when trying to add an account. So I removed my last account to start from scratch. All I get now from the mail app is a white screen with the ownCloud title bar. No errors in the development console - simply nothing?!?  Any idea?

**Comment by ChristophWurst** [comment]
> Hi!
> 
> Downloaded latest nightly from 19 October and applied the diff from your PR. After logging in the mail app was still working with my two accounts. So I removed the first account and noticed nothing happens when trying to add an account. So I removed my last account to start from scratch. All I get now from the mail app is a white screen with the ownCloud title bar. No errors in the development console - simply nothing?!? Any idea?

Maybe old js files are cached. Try to reload the page with CTRL+F5 and then do a ``localStorage.clear()`` in your browser's dev console.

> nothing happens when trying to add an account

What exactly do you mean by that? Did you see the setup form? Could you send it?

**Comment by jancborchardt** [comment]
> no loading spinner is shown in the middle of all inputs while loading. I always thought it kind of looked odd.

Where is it shown now though? On the button?

**Comment by ChristophWurst** [comment]
>> no loading spinner is shown in the middle of all inputs while loading. I always thought it kind of looked odd.
> 
> Where is it shown now though? On the button?
> We put it on top of the inputs because they are disabled for sending anyway. It should be visible all the time though and not go out of view when scrolling.

No, there's no loading spinner at all at the moment, I simply removed it. I didn't know it was placed there intentionally – I'll add it back ;-)

This is what it looked before btw (on FF):
bildschirmfoto von 2015-10-20 09-01-23
If the spinner should be shown on top of all disabled inputs, I'd say it should at least be centered.

**Comment by ChristophWurst** [comment]
I also forgot to fill in default account name and email address from the user's personal settings.

**Comment by jancborchardt** [comment]
Yes yes my friend, not only delete everything. ;)

**Comment by ChristophWurst** [comment]
@jancborchardt our loading spinner is back :smile: 
I noticed that we had a blue »Connect« button, so I removed the primary class to make it consistent with all the other buttons.

File: templates/index.php [hunk 1]
```diff
@@ -20,6 +20,10 @@
 }
 ?>
 
+<div id="user-displayname"
+     style="display: none"><?php p(\OCP\User::getDisplayName(\OCP\User::getUser())); ?></div>
+<div id="user-email"
+     style="display: none"><?php p(\OCP\Config::getUserValue(\OCP\User::getUser(), 'settings', 'email', '')); ?></div>
```
Comment (ChristophWurst) [inline templates/index.php]: @DeepDiver1975 is there a better way of retrieving those settings? Something like an ajax endpoint for apps would be helpful.

**Comment by jancborchardt** [comment]
> I noticed that we had a blue »Connect« button, so I removed the primary class to make it consistent with all the other buttons.

That should also stay though, since it’s the primary action button. ;) Like Send and Reply.

**Comment by ChristophWurst** [comment]
Dang it. I changed it back. Now we should have the same look and feel as before. @jancborchardt can you verify that? :-)

Btw: should a primary button have the same blue tone as the OC title bar? Currently they are different.

**Comment by Gomez** [comment]
:+1: 
No issues found for me.

**Pull request merged**